core/arch/instr.h

```diff
@@ -1350,6 +1350,10 @@ instr_predicate_reads_srcs(dr_pred_type_t pred);
 bool
 instr_predicate_writes_eflags(dr_pred_type_t pred);
 
+DR_API
+/**
+ * Returns true iff \p pred denotes a conditional predicate.
```

Comment:
I would spell out what it means as it can be hard to fully describe in prose, and add "truly" or sthg:

Returns true iff \p pred denotes a truly conditional predicate: on all architectures, this excludes #DR_PRED_NONE.  On ARM it also excludes #DR_PRED_AL and #DR_PRED_OP; on AArch64, it also excludes #DR_PRED_AL and #DR_PRED_NV.